services/motion/builtin/replanner.go

```diff
@@ -11,40 +14,40 @@ type replanResponse struct {
 	replan bool
 }
 
+// replanFn is an alias for a function that will be polled by a replanner
+type replanFn func(context.Context, [][]referenceframe.Input, int) (bool, error)
```

Comment:
I'm somewhat skeptical of this; I wonder how long it will be before we need to add new params.

Reply:
Also, it does bake in having to do this a specific way. Expecially the lack of a `Plan` for something that appears to be generally applicable.

Reply:
I don't think this should bloat too much.  The reason for this being there are many more parameters accessible in practice, as the replanFns I am using here are methods on moveRequest.  This is effectively using the same closure style solution I had previously but making them not anonymous functions

The only parameters that should be added here are those that are in the scope of the moveAttempt in question